### PR TITLE
Gradient application fix

### PR DESCRIPTION
**Pull request by PragTob**
For me this fixes the display of stars with a gradient.

screenshot from 2015-04-19 21 38 38

Also fixes the inspect on internal app, on which I stumbled while investigating this.

Sadly, I can't figure out what needs to change around shape to make the gradient work :-/

### CODE REVIEW

File: shoes-swt/lib/shoes/swt/gradient.rb [hunk 1]
```diff
@@ -44,13 +44,8 @@ def apply_as_stroke(gc, dsl)
...
       def extract_dimensions(dsl)
-        if dsl.is_a?(Star)
-          left = dsl.element_left - dsl.element_width / 2.0
-          top  = dsl.element_top - dsl.element_height / 2.0
-        else
-          left = dsl.element_left
-          top  = dsl.element_top
-        end
+        left = dsl.element_left
+        top  = dsl.element_top
```
Comment (jasonrclark) [inline shoes-swt/lib/shoes/swt/gradient.rb]: Interesting that this had that effect... don't feel like I understand why, but I haven't gotten familiar with the gradient code.

Noticed too that I think we could just unroll the `extract_dimensions` method now that it's basically just using the `dsl.element_*` accessors with no real additional computation. Saves an array allocation :wink:

Comment (PragTob) [inline shoes-swt/lib/shoes/swt/gradient.rb]: yeah I guess it was from back in the days when each thing had its own dimensions and the one of star were slightly weird... at least that's what I tell myself when I hope I didn't break any important behavior!

Also good catch about unrolling, will do!

**Comment by PragTob** [comment]
Pushed update with unrolled version! :)

**Comment by jasonrclark** [comment]
Nice! Pretty sure the Travis build was unrelated to us--failed bundling. Rerunning, and will happily merge once it's there.

**Pull request merged**